File: CSharp/Generator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace CSharp
{
    static class Generator
    {
        public static List<Point> GeneratePoints(int minValue, int maxValue, int count)
        {
            List<Point> points = new List<Point>(count);
            Random random = new Random();

            for (int i = 0; i < count; i++)
            {
                double x = random.Next(minValue, maxValue) + Math.Round(random.NextDouble(), 2);
                double y = random.Next(minValue, maxValue) + Math.Round(random.NextDouble(), 2);
                points.Add(new Point(x, y));
            }

            return points;
        }
    }
}

File: CPP/Generator.cpp
#include <stdlib.h>
#include <iostream>
#include "Generator.h"
#include "time.h"

double** Generator::generatePoints(int minValue, int maxValue, int countPoints)
{
	double** points = new double* [countPoints];
	for (int i = 0; i < countPoints; i++)
	{
		points[i] = new double[2];
	}

	srand(time(NULL));
	double randValue;
	int precisionPoints = 2;

	for (int i = 0; i < countPoints; i++)
	{
		for (int j = 0; j < 2; j++)
		{
			randValue = rand() % (int)pow(10, precisionPoints);
			randValue = minValue + (randValue / pow(10, precisionPoints)) * (maxValue - minValue);

			points[i][j] = randValue;
		}
	}

	return points;
}

std::list<point>* Generator::generatePointsList(int minValue, int maxValue, int countPoints)
{
	std::list<point>* points = new std::list<point>();

	srand(time(NULL));
	double value, x, y;
	int precisionPoints = 2;

	for (int i = 0; i < countPoints; i++)
	{
		value = rand() % (int)pow(10, precisionPoints);
		x = minValue + (value / pow(10, precisionPoints)) * (maxValue - minValue);
		value = rand() % (int)pow(10, precisionPoints);
		y = minValue + (value / pow(10, precisionPoints)) * (maxValue - minValue);

		points->push_back(point(x, y));
	}

	return points;
}
